tests/django_pg_migration_tools/test_operations.py

```diff
@@ -1755,3 +1756,701 @@ def test_when_valid_constraint_and_alter_table_already_performed(self):
             ALTER TABLE "example_app_nullintfieldmodel"
             DROP CONSTRAINT "example_ap_int_field_59f69830a8";
         """)
+
+
+class TestSaferSaferAddFieldForeignKey:
+    app_label = "example_app"
+
+    @pytest.mark.django_db
+    def test_requires_atomic_false(self):
+        project_state = ProjectState()
+        project_state.add_model(ModelState.from_model(IntModel))
+        new_state = project_state.clone()
+        operation = operations.SaferAddFieldForeignKey(
+            model_name="intmodel",
+            name="char_model_field",
+            field=models.ForeignKey(CharModel, null=True, on_delete=models.CASCADE),
+        )
+        with pytest.raises(NotSupportedError):
+            with connection.schema_editor(atomic=True) as editor:
+                operation.database_forwards(
+                    self.app_label, editor, project_state, new_state
+                )
+
+    @pytest.mark.django_db(transaction=True)
+    def test_when_not_null(self):
+        project_state = ProjectState()
+        project_state.add_model(ModelState.from_model(IntModel))
+        new_state = project_state.clone()
+        operation = operations.SaferAddFieldForeignKey(
+            model_name="intmodel",
+            name="char_model_field",
+            field=models.ForeignKey(CharModel, null=False, on_delete=models.CASCADE),
+        )
+        with pytest.raises(
+            ValueError, match="Can't safely create a FK field with null=False"
+        ):
+            with connection.schema_editor(atomic=True) as editor:
```

Comment:
Is there a particular reason to test with `atomic=True` in this test?

Reply:
Nope, it should be `False` actually:
be6a3d8e0d7a87402bcd8a207420471973b85941